core/arch/interp.c

```diff
@@ -4196,6 +4223,12 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
         }
 #endif
     }
+    else if (dcontext->forged_exception_addr == bb->start_pc) {
+        /* Might have been cleared by client_process_bb. */
```

Comment:
I see.  This is messy.  It's not clear what the best thing to do is (as my first comment talks about) so just do something with a XXX comment and update the issue (or if you prefer file a new issue).